pyomo/contrib/community_detection/detection.py

```diff
@@ -168,7 +170,225 @@ def detect_communities(model, type_of_community_map='constraint', with_objective
+            node_list = [node for sublist in list_of_node_lists for node in sublist]
+
+            # Now, we will find the first community that a node appears in and color the node based on that community
+            # In numbered_community_map, certain nodes may appear in multiple communities, and we have chosen to give preference
+            # to the first community a node appears in
+            color_list = []
+            for node in node_list:
+                not_found = True
+                for community_key in numbered_community_map:
+                    if not_found and node in numbered_community_map[community_key][position]:
+                        color_list.append(community_key)
+                        not_found = False
+
+            # Note - there is no strong reason to choose spring layout; it just creates relatively clean graphs
+            if pos is None:  # The case where the user has not provided their own layout
+                pos = nx.spring_layout(model_graph)
+
+        # Define color_map
+        color_map = cm.get_cmap('viridis', len(numbered_community_map))
+
+        # Create the figure and draw the graph
+        fig = plt.figure()
+        nx.draw_networkx_nodes(model_graph, pos, nodelist=node_list, node_size=40, cmap=color_map, node_color=color_list)
+        nx.draw_networkx_edges(model_graph, pos, alpha=0.5)
+
+        # Make the main title
+        graph_type = type_of_graph.capitalize()
+        community_map_type = self.type_of_community_map.capitalize()
+        main_graph_title = "%s graph - colored using %s community map" % (graph_type, community_map_type)
+
+        main_font_size = 14
+        plt.suptitle(main_graph_title, fontsize=main_font_size)
+
+        # Define a dict that will be used for the graph subtitle
+        subtitle_naming_dict = {'bipartite': 'Nodes are variables and constraints & Edges are variables in a constraint',
+                                'constraint': 'Nodes are constraints & Edges are common variables',
+                                'variable': 'Nodes are variables & Edges are shared constraints'}
+
+        # Make the subtitle
+        subtitle_font_size = 11
+        plt.title(subtitle_naming_dict[type_of_graph], fontsize=subtitle_font_size)
+
+        # Return the figure and pos, the position dictionary used for the graph layout
+        return fig, pos
+
+    def stringify(self):
+        """
+        This function takes in a community map of Pyomo components and returns the same community map but with the strings
+        of the Pyomo components. Alternatively, if the user does not provide a community map, this function can take in
+        a model and then create a community map (using the function detect_communities). Then, the same process of
+        "stringifying" the community map is performed, again resulting in a community map but with the strings
+        of the Pyomo components.
```

Comment:
Documentation seems a bit out of date, but you might choose to update it once the API is solidified.

On this function, does it make sense to override `__str__`?

Reply:
Yeah, sorry about the delay with the documentation. As for your second point, I was considering the following:
- remove the above function and put that code into the __str__ function so that when the user prints the CommunityMap, they will see a user-friendly community map (a bunch of string variable names in the dict)
- for __repr__, return the community_map attribute of the CommunityMap object (which would be the community map of actual Pyomo components)

Not sure if this answers your question, but I haven't made any changes yet as I'm unsure what would be best - let me know what you think!